### PR TITLE
[TensorFilter] Combined Tensor Picking @open sesame 11/05 14:43 

### PR DESCRIPTION
**Pull request by gichan-jang**
Add options for input and output combinations.

example pipeline
```
gst-launch-1.0 \
    videotestsrc ! videoconvert ! videoscale ! video/x-raw,width=640,height=480,format=RGB,framerate=30/1 ! tensor_converter ! mux.sink_0 \
    v4l2src ! videoconvert ! videoscale ! video/x-raw,width=640,height=480,format=RGB,framerate=30/1 ! tee name=t \
      t. ! queue leaky=2 max-size-buffers=2 ! videoscale ! video/x-raw,width=300,height=300,format=RGB ! tensor_converter ! \
        tensor_transform mode=arithmetic option=typecast:float32,add:-127.5,div:127.5 ! mux.sink_1 \
        tensor_mux name=mux ! tensor_filter framework=tensorflow-lite model=tflite_model/ssd_mobilenet_v2_coco.tflite inputCombination=1 outputCombination=i0,o0,o1 ! \
        tensor_demux name=demux \
        demux.src_0 ! queue leaky=2 max-size-buffers=10 ! tensor_decoder mode=direct_video ! videoconvert ! ximagesink sync=false async=false \
        demux.src_1 ! queue leaky=2 max-size-buffers=2 ! fmux.sink_0 \
        demux.src_2 ! queue leaky=2 max-size-buffers=2 ! fmux.sink_1 \
        tensor_mux name=fmux ! \
        tensor_decoder mode=bounding_boxes option1=tflite-ssd option2=tflite_model/coco_labels_list.txt option3=tflite_model/box_priors.txt option4=640:480 option5=300:300 ! \
        compositor name=mix sink_0::zorder=2 sink_1::zorder=1 ! videoconvert ! ximagesink \
      t. ! queue leaky=2 max-size-buffers=10 ! mix.
```

After #2847 (tensor_demux with other/tensors out)
```
gst-launch-1.0 \
    videotestsrc ! videoconvert ! videoscale ! video/x-raw,width=640,height=480,format=RGB,framerate=30/1 ! tensor_converter ! mux.sink_0 \
    v4l2src ! videoconvert ! videoscale ! video/x-raw,width=640,height=480,format=RGB,framerate=30/1 ! tee name=t \
      t. ! queue leaky=2 max-size-buffers=2 ! videoscale ! video/x-raw,width=300,height=300,format=RGB ! tensor_converter ! \
        tensor_transform mode=arithmetic option=typecast:float32,add:-127.5,div:127.5 ! mux.sink_1 \
        tensor_mux name=mux ! tensor_filter framework=tensorflow-lite model=tflite_model/ssd_mobilenet_v2_coco.tflite inputCombination=1 outputCombination=i0,o0,o1 ! \
        tensor_demux name=demux tensorpick=0,1+2 \
        demux.src_0 ! queue leaky=2 max-size-buffers=10 ! tensor_decoder mode=direct_video ! videoconvert ! ximagesink sync=false async=false \
        demux.src_1 ! queue leaky=2 max-size-buffers=2 ! tensor_decoder mode=bounding_boxes option1=tflite-ssd option2=tflite_model/coco_labels_list.txt option3=tflite_model/box_priors.txt option4=640:480 option5=300:300 ! \
        compositor name=mix sink_0::zorder=2 sink_1::zorder=1 ! videoconvert ! ximagesink \
      t. ! queue leaky=2 max-size-buffers=10 ! mix.
```

Related issue: #2803 

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_filter/tensor_filter_common.h [hunk 1]
```diff
@@ -103,6 +103,12 @@ typedef struct _GstTensorFilterPrivate
   gboolean is_updatable; /**<  a given model to the filter is updatable if TRUE */
   GstTensorsConfig in_config; /**< input tensor info */
   GstTensorsConfig out_config; /**< output tensor info */
+  GList *in_combi; /**< Select the input tensor(s) to invoke the models */
+  GList *out_combi_i; /**< Select the output tensor(s) from the input tensor(s) */
+  GList *out_combi_o; /**< Select the output tensor(s) from the model output */
+  gboolean in_combi_defined; /**< True if input combination is defined */
+  gboolean out_combi_i_defined;/**< True if output combination from input is defined */
+  gboolean out_combi_o_defined;/**< True if output combination from model output is defined */
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.h:111]: For readability, how about defining extra data structure for those variables? (e.g., `GstTensorConbination`)

**Review by myungjoo — APPROVED** [review]
LGTM. Write a complete manual entry for input/output Combination.

File: gst/nnstreamer/tensor_filter/tensor_filter_common.h (hunk 1 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.h]: Thank you for the suggestion. I agreed with your opinion. Updated.

**Comment by gichan-jang** [comment]
> LGTM. Write a complete manual entry for input/output Combination.

I will upload the manual. Thanks.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Pull request merged**